### PR TITLE
Update tracing to OpenTracing api 2

### PR DESCRIPTION
**Pull request by westover**

### CODE REVIEW

File: tchannel/tracing.py [hunk 1]
```diff
@@ -97,7 +97,11 @@ def span_in_context(self, span):
         :param span: an OpenTracing Span
         :return: ``StackContext``-based context manager
         """
-        return opentracing_instrumentation.span_in_stack_context(span)
+        if isinstance(opentracing.tracer.scope_manager,
+                      opentracing_instrumentation.request_context.TornadoScopeManager):  # noqa
+            return opentracing_instrumentation.span_in_stack_context(span)
+
+        return opentracing_instrumentation.span_in_context(span)
```
Comment (abhinav) [inline tchannel/tracing.py]: there doesn't appear to be a difference between the if and else cases. Is this a leftover from before?

Comment (westover) [reply tchannel/tracing.py]: span_in_stack_context vs span_in_context

File: tchannel/tracing.py [hunk 2]
```diff
@@ -97,7 +97,11 @@ def span_in_context(self, span):
         :param span: an OpenTracing Span
         :return: ``StackContext``-based context manager
         """
-        return opentracing_instrumentation.span_in_stack_context(span)
+        if isinstance(opentracing.tracer.scope_manager,
+                      opentracing_instrumentation.request_context.TornadoScopeManager):  # noqa
```
Comment (abhinav) [inline tchannel/tracing.py]: suspect you can make this pep8 friendly with

```
if isinstance(
    opentracing.tracer.scope_manager,
    opentracing_instrumentation.request_context.TornadoScopeManager,
):
    ...
```

Comment (westover) [reply tchannel/tracing.py]: Nope still too long :)

File: setup.py [hunk 3]
```diff
@@ -57,8 +57,8 @@
         'threadloop>=1,<2',
 
         # tracing deps
-        'opentracing>=1.1,<2',
-        'opentracing_instrumentation>=2.3,<3',
+        'opentracing>2',
+        'opentracing_instrumentation>3',
```
Comment (willsalz) [inline setup.py]: so…this now conflicts with jaeger-client-python, which requires opentracing_instrumentation<3

Comment (westover) [reply setup.py]: jaeger-client-python is internal to Uber so lets take this convo to uChat :)

**Pull request merged**